### PR TITLE
Fix some bugs

### PR DESCRIPTION
**Pull request by cyberpower678**
Add some comments.
Fix MOS
Add UTF-8 support
Fix bug that encoded = and & in queries and fragments.
Add some test cases.

### CODE REVIEW

**Comment by cyberpower678** [comment]
It would seem that my code got mangled after the conflict corruption dilemma.

**Comment by cyberpower678** [comment]
Something really weird is going on here.  I never touched those files it claims I'm committing.

File: src/CheckIfDead.php [hunk 1]
```diff
@@ -371,21 +375,66 @@ public function sanitizeURL( $url ) {
...
+			$url .= "#".$parts['fragment'];
 		}
 		return $url;
 	}
 
 	/**
+	 * Custom parse_url function to support UTF-8 URLs
+	 *
+	 * @param string $url The URL to parse. Invalid characters are replaced by _.
+	 * @param int $component Only return a given component
```
Comment (Niharika29) [inline src/CheckIfDead.php]: How is $component an int? Isn't it the PHP name for the component?

**Comment by cyberpower678** [comment]
They are ints.  The constants contain ints.

File: src/CheckIfDead.php (hunk 1 above)
Comment (cyberpower678) [inline src/CheckIfDead.php]: They are ints. The constants contain ints.

**Comment by Niharika29** [comment]
I'm not exactly happy merging this commit. You're rewriting a bunch of files for no reason. Can you try using a different editor and fixing this problem?

File: src/CheckIfDead.php [hunk 2]
```diff
@@ -332,15 +333,17 @@ protected function getDomainRoots( $url ) {
 	 * Properly encode the URL to ensure the receiving webservice understands the request.
 	 *
 	 * @param $url URL to sanitize
-	 * @return string sanitized URL
+	 * @return string sanitized URLs.  False on failure.
```
Comment (kaldari) [inline src/CheckIfDead.php]: Where does it return false on failure? I don't see it.

Comment (cyberpower678) [inline src/CheckIfDead.php]: I don't remember adding that.
It shouldn't return false.  Lemme check.

**Comment by Niharika29** [comment]
Looks good. Thanks for the PR!

**Pull request merged**